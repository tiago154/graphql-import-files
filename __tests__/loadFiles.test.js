const { loadFiles } = require('../src')
const { buildSchemaFromTypeDefinitions } = require('apollo-server')

describe('loadFiles', () => {
  describe('success', () => {
    it('should valid schema with many file', () => {
      const gropPattern = '**/__tests__/mocks/multifiles/**/*.{graphql,gql}'
      const schema = loadFiles(gropPattern)

      const schemaBuild = buildSchemaFromTypeDefinitions(schema)

      expect(schemaBuild).toMatchSnapshot()
    })

    it('should build the schema even with files with separate queries', () => {
      const gropPattern = '**/__tests__/mocks/separate-queries/*.graphql'
      const schema = loadFiles(gropPattern)

      const schemaBuild = buildSchemaFromTypeDefinitions(schema)

      expect(schemaBuild).toMatchSnapshot()
    })
  })

  describe('failure', () => {
    it('should returns an object with definitions empty when no file is founds', () => {
      const gropPattern = '**/__tests__/mocks/multifiles/**/*.{error,fail}'
      const schema = loadFiles(gropPattern)

      expect(schema).toEqual({ definitions: [], kind: 'Document' })
    })
  })
})