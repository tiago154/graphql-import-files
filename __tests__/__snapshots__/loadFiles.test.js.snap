// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadFiles success should build the schema even with files with separate queries 1`] = `
GraphQLSchema {
  "__allowedLegacyNames": Array [],
  "__validationErrors": undefined,
  "_directives": Array [
    "@skip",
    "@include",
    "@deprecated",
  ],
  "_implementations": Object {},
  "_mutationType": null,
  "_possibleTypeMap": Object {},
  "_queryType": "Query",
  "_subscriptionType": null,
  "_typeMap": Object {
    "Boolean": "Boolean",
    "Int": "Int",
    "Query": "Query",
    "String": "String",
    "User": "User",
    "__Directive": "__Directive",
    "__DirectiveLocation": "__DirectiveLocation",
    "__EnumValue": "__EnumValue",
    "__Field": "__Field",
    "__InputValue": "__InputValue",
    "__Schema": "__Schema",
    "__Type": "__Type",
    "__TypeKind": "__TypeKind",
  },
  "astNode": Object {
    "directives": Array [],
    "kind": "SchemaDefinition",
    "loc": Object {
      "end": 23,
      "start": 0,
    },
    "operationTypes": Array [
      Object {
        "kind": "OperationTypeDefinition",
        "loc": Object {
          "end": 21,
          "start": 9,
        },
        "operation": "query",
        "type": Object {
          "kind": "NamedType",
          "loc": Object {
            "end": 21,
            "start": 16,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 21,
              "start": 16,
            },
            "value": "Query",
          },
        },
      },
    ],
  },
  "extensionASTNodes": undefined,
  "extensions": undefined,
}
`;

exports[`loadFiles success should valid schema with many file 1`] = `
GraphQLSchema {
  "__allowedLegacyNames": Array [],
  "__validationErrors": undefined,
  "_directives": Array [
    "@skip",
    "@include",
    "@deprecated",
  ],
  "_implementations": Object {},
  "_mutationType": null,
  "_possibleTypeMap": Object {},
  "_queryType": "Query",
  "_subscriptionType": null,
  "_typeMap": Object {
    "Boolean": "Boolean",
    "Float": "Float",
    "ID": "ID",
    "Int": "Int",
    "Product": "Product",
    "Query": "Query",
    "String": "String",
    "User": "User",
    "__Directive": "__Directive",
    "__DirectiveLocation": "__DirectiveLocation",
    "__EnumValue": "__EnumValue",
    "__Field": "__Field",
    "__InputValue": "__InputValue",
    "__Schema": "__Schema",
    "__Type": "__Type",
    "__TypeKind": "__TypeKind",
  },
  "astNode": Object {
    "directives": Array [],
    "kind": "SchemaDefinition",
    "loc": Object {
      "end": 23,
      "start": 0,
    },
    "operationTypes": Array [
      Object {
        "kind": "OperationTypeDefinition",
        "loc": Object {
          "end": 21,
          "start": 9,
        },
        "operation": "query",
        "type": Object {
          "kind": "NamedType",
          "loc": Object {
            "end": 21,
            "start": 16,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 21,
              "start": 16,
            },
            "value": "Query",
          },
        },
      },
    ],
  },
  "extensionASTNodes": undefined,
  "extensions": undefined,
}
`;
